js/piwik.js

```diff
@@ -2449,7 +2449,7 @@ if (typeof window.Matomo !== 'object') {
                     ';SameSite=' + sameSite;
 
                 // check the cookie was actually set
-                if (getCookie(cookieName) !== value) {
+                if ((!msToExpire || msToExpire >= 0) && getCookie(cookieName) !== value) {
```

Comment:
should it be `msToExpire >= 1` or `>= 5` maybe as the value might be expired by the time it's set and read? I suppose we don't set such values so it doesn't really matter.

Reply:
Guess no one should use such low values for expire time. If someone does nevertheless, might be good to have a warning, as it simply might not make any sense to set a cookie that expires before it can be used 🙈